File: app/src/main/java/com/centaury/cataloguemovie/data/local/db/CatalogueDatabase.java
package com.centaury.cataloguemovie.data.local.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.centaury.cataloguemovie.data.local.entity.MovieEntity;
import com.centaury.cataloguemovie.data.local.entity.TVShowEntity;

/**
 * Created by Centaury on 11/20/2019.
 */
@Database(entities = {MovieEntity.class, TVShowEntity.class}, version = 1, exportSchema = false)
public abstract class CatalogueDatabase extends RoomDatabase {

    private static CatalogueDatabase INSTANCE;

    public abstract CatalogueDao catalogueDao();

    private static final Object sLock = new Object();

    public static CatalogueDatabase getInstance(Context context) {
        synchronized (sLock) {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                        CatalogueDatabase.class, "Catalogue.db")
                        .build();
            }
            return INSTANCE;
        }
    }
}

File: data/src/main/java/com/centaury/data/db/CatalogueDatabase.kt
package com.centaury.data.db

import androidx.room.Database
import androidx.room.RoomDatabase
import com.centaury.data.db.dao.MovieDao
import com.centaury.data.db.dao.TVShowDao
import com.centaury.data.movies.repository.source.local.entity.MovieEntity
import com.centaury.data.tvshow.repository.source.local.entity.TVShowEntity

/**
 * @Author Centaury (alfa.arnialfa@gmail.com)
 * Created by Centaury on 7/24/2020.
 */
@Database(entities = [MovieEntity::class, TVShowEntity::class], version = 1, exportSchema = false)
abstract class CatalogueDatabase : RoomDatabase() {

    abstract fun movieDao(): MovieDao

    abstract fun tvShowDao(): TVShowDao

    companion object {
        const val DATABASE_NAME = "movie_catalogue"
    }
}